### PR TITLE
chore: refactor trial logs CLI code and remove old api endpoints

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change removes `/trials/:id/logs` and `/trials/:id/logsv2` and refactors the CLI code in a way necessary to the change.

## Test Plan
- [x] test all log functionality still works.

## Commentary (optional)
Wow this CLI code was convoluted.. I think it's much better now.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: common/determined_common/api/experiment.py [hunk 1]
```diff
@@ -103,10 +121,7 @@ def follow_experiment_logs(master_url: str, exp_id: int) -> None:
     print("Following first trial with ID {}".format(first_trial_id))
 
     # Call `logs --follow` on the new trial.
```
Comment (dzhu) [inline common/determined_common/api/experiment.py:123]: I guess this comment is not as applicable as it was before.

**Review by dzhu — APPROVED** [review]
Thanks for cleaning this up!

File: master/internal/core_trial.go [hunk 2]
```diff
@@ -65,50 +64,6 @@ func (m *Master) getTrialMetrics(c echo.Context) (interface{}, error) {
...
-		TrialID       int  `path:"trial_id"`
-		GreaterThanID *int `query:"greater_than_id"`
-		LessThanID    *int `query:"less_than_id"`
-		Limit         *int `query:"tail"`
-	}{}
-	if err := api.BindArgs(&args, c); err != nil {
-		return err
-	}
-
-	logs, err := m.db.TrialLogsRaw(args.TrialID, args.GreaterThanID, args.LessThanID, args.Limit)
```
[the remark this answers is not in the document]
Comment (stoksc) [reply master/internal/core_trial.go:79]: ah of course, i got distracted by how annoying the CLI bit was by the time i got here. thanks for the catch.

File: common/determined_common/api/experiment.py [hunk 3]
```diff
@@ -42,48 +54,54 @@ def to_levels_above(level: str) -> List[str]:
...
+    if rank_ids is not None:
+        query["rank_ids"] = rank_ids
+    if sources is not None:
+        query["sources"] = sources
+    if stdtypes is not None:
+        query["stdtypes"] = stdtypes
+    if timestamp_before is not None:
+        query["timestamp_before"] = timestamp_before
+    if timestamp_after is not None:
+        query["timestamp_after"] = timestamp_after
```
[the remark this answers is not in the document]
Comment (stoksc) [reply common/determined_common/api/experiment.py:81]: lgtm, thanks

File: cli/determined_cli/trial.py [hunk 4]
```diff
@@ -104,6 +104,25 @@ def kill_trial(args: Namespace) -> None:
     print("Killed trial {}".format(args.trial_id))
 
 
+@authentication_required
+def follow_trial_logs(args: Namespace) -> None:
```
[the remark this answers is not in the document]
Comment (stoksc) [reply cli/determined_cli/trial.py:108]: ah yeah, true.

**Pull request merged**